src/iosrv/tests/drpc_client_tests.c

```diff
@@ -69,6 +69,18 @@ dss_self_rank(void)
 	return (d_rank_t)mock_self_rank;
 }
 
+struct dss_module_info	*mock_dmi = NULL;
+static uint32_t		 mock_xs_id = 1;
+struct dss_module_info	*
+get_module_info(void)
+{
+	D_ALLOC(mock_dmi, sizeof(struct dss_module_info));
+
+	mock_dmi->dmi_xs_id = mock_xs_id;
+
+	return mock_dmi;
+}
```

Comment:
Not able to mock dss_get_module_info() directly?

Reply:
no unfortunately not, it is static in line in daos_server.h and calls methods from tls.c and srv.c, creating a shadow function ended up being the nicest solution I could find without creating too much of a ripple effect, maybe using weak symbols could work as suggested by @jolivier23 but because a number of the relevant functions are static in line it looks like that might not work

Reply:
Ah, okay. This solution works well enough. Thanks for the explanation.